server/apps/archive/archive_crop.py

```diff
@@ -0,0 +1,164 @@
+        # Check if type is picture
+        if original.get('type') != 'picture':
+            raise SuperdeskApiError.badRequestError(message='Only images can be cropped!')
+
+        # Check if the renditions exists
+        if not original.get('renditions'):
+            raise SuperdeskApiError.badRequestError(message='Missing renditions!')
+
+        # Check if the original rendition exists
+        if not original.get('renditions').get('original'):
+            raise SuperdeskApiError.badRequestError(message='Missing original rendition!')
+
+        # Check if the crop name is valid
+        crops = get_resource_service('vocabularies').find_one(req=None, _id='crop_sizes')['items']
+        crop = self._get_crop_by_name(crops, crop_name)
+        if not crop:
+            raise SuperdeskApiError.badRequestError(message='Unknown crop name!')
+
+        self._validate_aspect_ratio(crop, doc)
+
+    def _validate_aspect_ratio(self, crop, doc):
+        """
+        Checks if the aspect ratio is consistent with one in defined in spec
+        :param crop: Spec parameters
+        :param doc: Posted parameters
+        :raises SuperdeskApiError.badRequestError:
+        """
+        width = doc['CropRight'] - doc['CropLeft']
+        height = doc['CropBottom'] - doc['CropTop']
+        doc_ratio = round(width / height, 1)
+        spec_ratio = round(crop['width'] / crop['height'], 1)
+        if doc_ratio != spec_ratio:
+            raise SuperdeskApiError.badRequestError(message='Wrong aspect ratio!')
+
+    def _get_crop_by_name(self, crops, crop_name):
+        """
+        Finds the crop in the list of crops by name
+        :param crops: List of crops
+        :param crop_name: Crop name
+        :return: Matching crop or None
+        """
+        try:
+            matching_crops = list(filter(lambda crop: crop.get('name', '').lower() == crop_name.lower(), crops))
+            if len(matching_crops) > 0:
+                return matching_crops[0]
+            return None
+        except:
+            return None
+
+    def _delete_crop(self, original, crop_name):
+        """ Deletes an existing crop with the given name """
+        renditions = original.get('renditions')
+        renditions.pop(crop_name, None)
+        get_resource_service('archive').patch(original['_id'], {'renditions': renditions})
+
+    def _add_crop(self, original, crop_name, doc):
+        """
+        Updates the crop with the same name if exists,
+        otherwise creates a new crop with the given parameters
+        :param original: Article to add the crop
+        :param crop_name: Name of the crop
+        :param doc: Crop details
+        :raises SuperdeskApiError.badRequestError
+        """
+        renditions = original.get('renditions')
+        crop_data = UploadService().get_cropping_data(doc)
+        original_image = original.get('renditions').get('original')
+        file = superdesk.app.media.fs(None).find_one(original_image.get('media'))
```

Comment:
Just wondering if it is better to use a different variable name than "file", despite this being Python 3.x? `file` is a built-in in Python 2 and that might be a bit confusing for random contributors that mostly work in Python 2.